cocos/platform/android/CCFileUtils-android.cpp

```diff
@@ -280,6 +280,45 @@ long FileUtilsAndroid::getFileSize(const std::string& filepath)
     return size;
 }
 
+std::vector<std::string> FileUtilsAndroid::listFiles(const std::string& dirPath) const
+{
+    std::vector<std::string> fileList;
+    string fullPath = fullPathForFilename(dirPath);
+
+    static const std::string apkprefix("assets/");
+
+    string relativePath = "";
+    size_t position = fullPath.find(apkprefix);
+    if (0 == position) {
+        // "assets/" is at the beginning of the path and we don't want it
+        relativePath += fullPath.substr(apkprefix.size());
+    } else {
+        relativePath = fullPath;
+    }
+
+    if (nullptr == assetmanager) {
+        LOGD("... FileUtilsAndroid::assetmanager is nullptr");
+        return fileList;
+    }
+
+    auto *dir = AAssetManager_openDir(assetmanager, relativePath.c_str());
+    if(nullptr == dir) {
+        LOGD("... FileUtilsAndroid::failed to open dir %s", relativePath.c_str());
+        AAssetDir_close(dir);
+        return fileList;
+    }
+    const char *tmpDir = nullptr;
+    while((tmpDir = AAssetDir_getNextFileName(dir))!= nullptr)
+    {
+        string filepath(tmpDir);
+        if(isDirectoryExistInternal(filepath)) filepath += "/";
```

Comment:
Why using `isDirectoryExistInternal`? This function is to find a file, not a directory.

Reply:
default implementation includes directories in result 
https://github.com/cocos2d/cocos2d-x/blob/3cad2a85d49fbd787c0c68e960fe0678b28bbc09/cocos/platform/CCFileUtils.cpp#L1453-L1459